src/callbacks/liquidity/Ramses/RamsesV2DTL.sol

```diff
@@ -0,0 +1,239 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+// Libraries
+import {ERC20} from "@solmate-6.7.0/tokens/ERC20.sol";
+
+// Callbacks
+import {BaseDirectToLiquidity} from "../BaseDTL.sol";
+
+// Ramses
+import {IRamsesV2Pool} from "./lib/IRamsesV2Pool.sol";
+import {IRamsesV2Factory} from "./lib/IRamsesV2Factory.sol";
+import {IRamsesV2PositionManager} from "./lib/IRamsesV2PositionManager.sol";
+
+// Uniswap
+import {TickMath} from "@uniswap-v3-core-1.0.1-solc-0.8-simulate/libraries/TickMath.sol";
+import {SqrtPriceMath} from "../../../lib/uniswap-v3/SqrtPriceMath.sol";
+
+/// @title      RamsesV2DirectToLiquidity
+/// @notice     This Callback contract deposits the proceeds from a batch auction into a Ramses V2 pool
+///             in order to create full-range liquidity immediately.
+///
+///             The LP tokens are transferred to `DTLConfiguration.recipient`, which must be an EOA or a contract that can receive ERC721 tokens.
+///
+///             An important risk to consider: if the auction's base token is available and liquid, a third-party
+///             could front-run the auction by creating the pool before the auction ends. This would allow them to
+///             manipulate the price of the pool and potentially profit from the eventual deposit of the auction proceeds.
+///
+/// @dev        As a general rule, this callback contract does not retain balances of tokens between calls.
+///             Transfers are performed within the same function that requires the balance.
+contract RamsesV2DirectToLiquidity is BaseDirectToLiquidity {
+    // ========== ERRORS ========== //
+
+    error Callback_Params_PoolFeeNotEnabled();
+
+    // ========== STRUCTS ========== //
+
+    /// @notice     Parameters for the onSettle callback
+    /// @dev        This will be encoded in the `callbackData_` parameter
+    ///
+    /// @param      maxSlippage             The maximum slippage allowed when adding liquidity (in terms of `ONE_HUNDRED_PERCENT`)
+    /// @param      veRamTokenId            The token ID of the veRAM token to use for the position (optional)
+    struct OnSettleParams {
```

Comment:
These parameters need to be set during onCreate instead of onSettle. `settle` can be called by anyone and they could input malicious values for slippage or token ID.

Reply:
Done